packages/react-app/src/pages/api/bounties/[id].ts

```diff
@@ -1,90 +1,68 @@
+			key,
+		});
+	}
 	await dbConnect();
 
-	switch (method) {
+	const bounty = await service.getBounty(id as string);
+	if (!bounty) {
+		return notFound(res);
+	}
+
+	switch (req.method) {
 	case 'GET':
 		/* Get a model by its ID */
-		try {
-			if (id.length !== 24) {
-				return res.status(404).json({ success: false });
-			}
-			const bounty = await Bounty.findById(id).exec();
-			if (!bounty) {
-				return res.status(404).json({ success: false });
-			}
-			res.status(200).json({ success: true, data: bounty });
-		} catch (error) {
-			res.status(400).json({ success: false });
-		}
+		res.status(200).json({ success: true, data: bounty });
 		break;
 
-	case 'PUT' :
+	case 'PATCH' :
 		/* Edit a model by its ID */
 		try {
-			const bounty: BountyCollection = await Bounty.findById(id).exec();
-			if (bounty.editKey !== key) {
-				return res.status(400).json({ success: false });
-			}
-			if (
-				bounty.status.toLowerCase() === 'draft' ||
-          bounty.status.toLowerCase() === 'open'
-			) {
-				await Bounty.findByIdAndUpdate(id, req.body, {
-					new: true,
-					omitUndefined: true,
-					runValidators: true,
-				}).then(async (updateBounty: BountyCollection) => {
-					await publishBountyToDiscordChannel(updateBounty, bounty.status);
-					res.status(200).json({ success: true, data: updateBounty });
-				}).catch(() => {
-					return res.status(400).json({ success: false });
+			const bountyIsEditable = service.canBeEdited({ bounty, key });
+			if (!bountyIsEditable) {
+				return res.status(400).json({
+					success: false,
+					message: 'Unable to edit bounty, either the edit key is invalid or it is not in an editable status',
+					key,
+					bountyStatus: bounty.status,
 				});
-			} else {
-				return res.status(400).json({ success: false });
 			}
+			const updateBounty = await service.editBounty({ bounty, body: req.body });
+			res.status(200).json({ success: true, data: updateBounty });
 		} catch (error) {
-			res.status(400).json({ success: false });
+			internalServerError(res);
 		}
 		break;
-	default:
-		res.status(400).json({ success: false });
+		
+	case 'DELETE':
+		try {
+			await service.deleteBounty(id as string);
+			res.status(204).end();
+		} catch (error) {
+			res.status(400).json({ success: false, error });
+		}
 		break;
-	}
-}
 
-export const publishBountyToDiscordChannel = async (
-	bounty: BountyCollection,
-	previousStatus: string
-): Promise<any> => {
-	if (previousStatus.toLowerCase() !== 'draft') {
-		return;
+	default:
+		return internalServerError(res);
 	}
-	const embedMessage = DiscordUtils.generateBountyEmbedsMessage(bounty);
-	return await fetch(BOUNTY_BOARD_WEBHOOK_URI + '?wait=true', {
-		method: 'POST',
-		headers: {
-			'Content-Type': 'application/json',
-		},
-		body: JSON.stringify(embedMessage),
-	})
-		.then((response) => {
-			if (response.status !== 200) console.log(response);
-		})
-		.catch(console.error);
 };
+
+export default middlewares({ schema: BountySchema, handler });
```

Comment:
I find this confusing. What about calling the validation and auth handers directly? I think it's chaining that breaks an expected pattern, I.e
```
await validate(handler)
await auth(handler)
await activity(handler)
```
vs validate(auth(activity(handler)))